### PR TITLE
Single file delete

### LINKED ISSUE #104: Remove individual files from source control
As noted in #80, in cases where entire packages are in source control, individual files do not get removed from source control. 

To get around this, we should add all the files in the package to source control instead of the package itself. 

This makes removing packages from source control a little more complex.

### PR DESCRIPTION
**Pull request by isc-svelury**
Removing a single file from source control when the whole package is being tracked now works as expected. 

1. Each file is tracked individually and all these files get added/removed automatically if you add/remove a package from source control. 
2. If you remove a file from source control and its parent package(s) is being tracked, it is removed automatically. Note, this will trigger a chain of parent package/project/csp folder removals till we no have parents left. 

I have covered everything to the best of knowledge. Please double check this PR in case I missed something. Thanks!

Note: This PR has commits from #110. Please merge that before this. Sorry for the confusion. 

Closes #104

### CODE REVIEW

**Review by isc-tleavitt — COMMENTED** [review]
Some reasonable scope efficiency improvements, then I'll kick the tires on it a little.

File: cls/SourceControl/Git/Utils.cls [hunk 1]
```diff
@@ -444,12 +445,107 @@ ClassMethod DeleteExternalsForItem(InternalName As %String) As %Status
...
+ClassMethod FindTrackedFilesInPackage(InternalName As %String, ByRef trackedFiles As %String) As %Status
+{
+    Set sc = $$$OK
+    set item = ""
+    set trackedFiles = ""
+    for{
+        set item = $order(@..#Storage@("items",item))
+        quit:item=""
+        
+        if ..IsClassInPackage(item, InternalName) {
```
[the remark this answers is not in the document]
Comment (isc-svelury) [reply cls/SourceControl/Git/Utils.cls:470]: Addressed in the latest commit

File: cls/SourceControl/Git/Utils.cls [hunk 2]
```diff
@@ -444,12 +445,107 @@ ClassMethod DeleteExternalsForItem(InternalName As %String) As %Status
...
+ClassMethod FindTrackedFilesInProjects(InternalName As %String, ByRef trackedFiles As %String) As %Status
+{
+    Set sc = $$$OK
+    #dim item as %String = ""
+    set trackedFiles = ""
+    for{
+        set item = $order(@..#Storage@("items",item))
+        quit:item=""
+
+        if ..IsItemInProject(item, InternalName) {
```
[the remark this answers is not in the document]
Comment (isc-svelury) [reply cls/SourceControl/Git/Utils.cls:491]: Did not address in the latest commit; discussed why on Teams.

File: cls/SourceControl/Git/Utils.cls [hunk 3]
```diff
@@ -444,12 +445,107 @@ ClassMethod DeleteExternalsForItem(InternalName As %String) As %Status
...
+ClassMethod FindTrackedFilesInCSPFolders(InternalName As %String, ByRef trackedFiles As %String) As %Status
+{
+    Set sc = $$$OK    
+    #dim item as %String = ""
+    set trackedFiles = ""
+    for  {
+        set item = $order(@..#Storage@("items",item))
+        quit:item=""
+        
+        if ..IsItemInCSPFolder(item, InternalName) {
```
[the remark this answers is not in the document]
Comment (isc-svelury) [reply cls/SourceControl/Git/Utils.cls:512]: Addressed in the latest commit

**Review by isc-tleavitt — APPROVED** [review]
Looks good!

**Pull request merged**